.github/workflows/post_release_version_bump.yml

```diff
@@ -102,6 +101,25 @@ jobs:
           sed -i 's/python:v.*"/python:v'$VERSION'"/' .github/workflows/daily_scan.yml
           git add aws-opentelemetry-distro/src/amazon/opentelemetry/distro/version.py
           git add .github/workflows/daily_scan.yml
+  
+      - name: Append latest release checksum to release-build-metadata.json
+        run: |
+          ARTIFACT_NAME="aws_opentelemetry_distro-${{ github.event.inputs.version }}-py3-none-any.whl"
+          curl -L -o $ARTIFACT_NAME https://github.com/aws-observability/aws-otel-python-instrumentation/releases/download/v${{ github.event.inputs.version }}/$ARTIFACT_NAME
+
+          CHECKSUM=$(shasum -a 256 $ARTIFACT_NAME | awk '{ print $1 }')
+
+          FILE="release-build-metadata.json"
+          UPDATED_JSON=$(jq --arg version "${{ env.VERSION }}" \
+                            --arg name "$ARTIFACT_NAME" \
+                            --arg checksum "$CHECKSUM" \
+                            '.release += [{"version": $version, "checksum": [{"name": $name, "checksum": $checksum}]}]' "$FILE")
```

Comment:
Do we really need to append new version entry to the metadata file or can we just overwrite with the latest one? Do we have a use case of maintaining the metadata for older versions? 
With time, the file will grow big and more expensive to parse.

Reply:
Thinking a little more on this. If we do need to maintain the checksum for each released version, would storing the checksum file alongside the artifact within the GH release (like [this one](https://github.com/aws-observability/aws-otel-python-instrumentation/releases/tag/v0.7.0)) itself work?
This way, any user can use this checksum to validate the integrity of the artifact.

Reply:
> Do we really need to append new version entry to the metadata file or can we just overwrite with the latest one? Do we have a use case of maintaining the metadata for older versions? With time, the file will grow big and more expensive to parse.

That's a good point. I decided to maintain the metadata for the older versions since customers may still be using them and the older versions are still available even if we release new ones so thought it's better to have a record of them. I'm neutral if we want to just overwrite with the new ones though



> Thinking a little more on this. If we do need to maintain the checksum for each released version, would storing the checksum file alongside the artifact within the GH release (like [this one](https://github.com/aws-observability/aws-otel-python-instrumentation/releases/tag/v0.7.0)) itself work? This way, any user can use this checksum to validate the integrity of the artifact.

That was a consideration we made and we decided to go with the `.json` file because it's safer to retrieve programmatically. Also for cases like Dotnet where there is multiple artifacts, didn't want to overfill the release page with just checksum values